cmd/kubelet/app/server.go

```diff
@@ -440,6 +446,132 @@ func InitializeTLS(s *options.KubeletServer) (*server.TLSOptions, error) {
 	return tlsOptions, nil
 }
 
+// createCertAndKey tries to:
+// Use the boostrap auth token to get x509 certs if the token is set.
+// If this fails or the token is empty, then self-generates cert and key pair.
+func createCertAndKey(s *options.KubeletServer) error {
+	var err error
+	if s.BootstrapAuthToken != "" {
```

Comment:
is this just used to authenticate to the API? the kubelet already has credentials to talk to the API, provided as `--kubeconfig`... why not use those instead?